src/components/MarqueeSelection/MarqueeSelection.Props.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { ISelection } from '../../utilities/selection/interfaces';
+import { MarqueeSelection } from './MarqueeSelection';
+
+export interface IMarqueeSelectionProps extends React.Props<MarqueeSelection> {
+  /**
+   * The selection object to interact with when updating selection changes.
+   */
+  selection: ISelection;
+
+  /**
+   * The base element tag name to render the marquee bounding area within.
+   * @default div
+   */
+  rootTagName?: string;
+
+  /**
+   * Optional props to mix into the root element.
+   */
+  rootProps?: React.HTMLProps<HTMLDivElement>;
+
+  /**
+   * Optional callback that is called, when the mouse down event occurs, in order to determine
+   * if we should start a marquee selection. If true is returned, we will cancel the mousedown
+   * event to prevent upstream mousedown handlers from executing.
+   */
+  onShouldStartSelection?: (ev: React.MouseEvent) => boolean;
```

Comment:
I think this should be widened or altered.
To control whether or not marquee selection is allowed at all, I think passing an `isEnabled` flag makes more sense. However, to control whether or not a current drag gesture should activate a marquee, I think you should pass the proposed initial selection `Rectangle` so the owner can decide whether to let selection continue. The `MouseEvent` by itself is insufficient.

``` typescript
@bind
private _onShouldStartSelection(area: Rectangle, event: React.MouseEvent): boolean {
    if (!this._isDropping) {
        return false;
    }

    if (area.width > 10 || area.height > 10) {
        return true;
    }

    return false;
}
```